### PR TITLE
adds Matcher classes and API endpoint for matching

### PR DESCRIPTION
**Pull request by BrokenShell**
# Adds the following...
- 4 Matcher Models
    - MatcherRandom
    - MatcherSearch
    - MatcherSort
    - MatcherSortSearch
- API endpoint for matching
- Updates requirements.txt
- Random Generator Package

## Contributors
**Robert Sharp**
Kevin Lynner
Ryland Gomez
Rodrico Benito Sanchez
Michael Tran
Samuel Ulloa
mohamed mosaed
Archana Coimbatore Sivagurunathan
Olatomi Adigun
ZacharyRock
SirivennelaVempati
BhavaniLakshmi Annapurna Jagarlamudi
Chunyan "Christine" Wang

## PR Type
- [x] New feature

## Checklist:
- [x] My code follows PEP8 style guide
- [x] My changes generate no errors
- [x] No commented-out code
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by RylandGomez — APPROVED** [review]
As far as I can tell, the code is sound and follows PEP8. The added functions allow for the easy creation of mock data as well as expands and cleans up preexisting CRUD operations. Also very helpful that it introduces an MVP for a matching algorithm and corresponding endpoint that the web backend can eventually access.
I was gonna criticize data_generators being put into production until I saw it in .ebignore. Nice of you to integrate the preexisting notebook work into data_options.py file, as well. Failed to find any solid critiques, so full approval from me!

**Comment by ChristineWangcy** [comment]
This looks good. Everything is standard.

**Review by ChristineWangcy — APPROVED** [review]
It looks good. Everything is standard.

**Review by Annapurnaj91 — APPROVED** [review]
Everything looks great. So many things to learn especially the algorithms.

**Review by sirivennelavempati — APPROVED** [review]
Code was nicely written using the Polymorphism technique and there are so many techniques to learn.

**Review by archana062031 — APPROVED** [review]
Good example of Polymorphism.

**Comment by mktran0417** [comment]
Perhaps this is more pedantic, but is there any reason as to why the path parameters aren't at the end of the route? To me that seems to make a bit more sense but that could just be because of convention.

**Review by rodricobsanchez — APPROVED** [review]
everything follows pep8 and is simple to read and make future changes on to

**Review by OlatomiAdigun — APPROVED** [review]
Conforms to PEP8 standard. Great code, in all.

File: data_generators/random_match.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+def rand_lib(n_matches: int) -> List[str]:
+    return [match["user_id"] for match in sample(data, k=n_matches)]
+
+
+def pandas_lib(n_matches: int) -> List[str]:
+    return DataFrame(data).sample(n_matches)["user_id"].to_list()
+
+
+if __name__ == '__main__':
```
Comment (ulloa) [inline data_generators/random_match.py:19]: Do we need `if __name__ == '__main':` code block in our data generators and matching? I think the code could be handed over to members working on creating tests. I believe there should be a main script call for the data generation, but I think this doesn't do that.

**Review by ulloa — COMMENTED** [review]
Why did you delete experiment.ipynb in this pull request? Also, I had a comment on the `if __name__ == '__main__':` blocks. I think we could hand that code over to the members working on tests. It doesn't hurt to leave them in for now

**Pull request merged**